### DISCUSSION TITLE
Shifting from Jest to Vitest

### DISCUSSION THREAD

**Discussion by vhscom** [rgossiaux/svelte-headlessui#67] [Ideas]
https://github.com/nickbreaton/vitest-svelte-kit makes it deadly simple to get set-up with https://vitest.dev and start running unit tests. Will you consider shifting away from Jest to Vitest as the library matures? It seems there's an opportunity to shed some baggage given Jest was built around React and Vitest has a more focused integration with Vite.

**Comment by vhscom** [comment]
Here's a migration guide: https://vitest.dev/guide/migration.html#migrating-from-jest

**Comment by rgossiaux** [comment]
Yeah, I'm definitely excited about Vitest. I spent maybe an hour last month trying to migrate this project to use it but was unsuccessful; I couldn't get the import resolution to work correctly. I ultimately gave up since there's not really much benefit to migrating (the current setup works just fine) but I might revisit once Vitest matures some more.

Thanks for the repository link, I may give that a shot the next time I try this (or if I try in a personal project).

**Reply by vhscom** [reply]
Awesome. I just submitted a request against kit so if there's any movement on that I imagine integration will become easier and the rough edges smoothed out sooner: https://github.com/sveltejs/kit/issues/4143

**Reply by rgossiaux** [reply]
I opened a similar one for svelte-add here: https://github.com/svelte-add/svelte-add/issues/191
